src/lib/assets/yaml.ts

```diff
@@ -1,19 +1,42 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: 2023-Present The Pepr Authors
 
-import { dumpYaml } from "@kubernetes/client-node";
-import crypto from "crypto";
+import {
+  dumpYaml,
+  V1Deployment,
+  V1MutatingWebhookConfiguration,
+  V1ValidatingWebhookConfiguration,
+} from "@kubernetes/client-node";
 import { promises as fs } from "fs";
-import { Assets } from ".";
 import { apiTokenSecret, service, tlsSecret, watcherService } from "./networking";
-import { getDeployment, getModuleSecret, getNamespace, getWatcher } from "./pods";
+import { getModuleSecret, getNamespace } from "./pods";
 import { clusterRole, clusterRoleBinding, serviceAccount, storeRole, storeRoleBinding } from "./rbac";
-import { webhookConfig } from "./webhooks";
 import { genEnv } from "./pods";
+import { ModuleConfig } from "../core/module";
+import { CapabilityExport } from "../types";
+import { TLSOut } from "../tls";
 
+type AssetsOverrides = {
+  apiToken: string;
+  capabilities: CapabilityExport[];
+  config: ModuleConfig;
+  hash: string;
+  image: string;
+  name: string;
+};
+
+type AssetsOverrides2 = {
+  apiToken: string;
+  capabilities: CapabilityExport[];
+  config: ModuleConfig;
+  hash: string;
+  name: string;
+  path: string;
+  tls: TLSOut;
+};
```

Comment:
Placeholder types, consider consolidation and alternatives. Used to just be typed as the `Assets` class.